javascript/operations.js

```diff
@@ -473,7 +487,8 @@ export default {
       const { title, options } = operation
       Notification.requestPermission().then(result => {
         operation.permission = result
-        if (result === 'granted') new Notification(title || '', options)
+        if (result === 'granted')
+          new Notification(title != null ? title : '', options)
```

Comment:
We should probably coerce `options` to an object for consistency.

Reply:
If something is passed in, that is not an object, it gives a JS error currently, as somewhat cryptic one:

```
[Error] CableReady detected an error in notification: undefined is not an object (evaluating 'Notification.requestPermission().then'). If you need to support older browsers make sure you've included the corresponding polyfills. https://docs.stimulusreflex.com/setup#polyfills-for-ie11.
	(anonymous function) (vendors-node_modules_hotwired_turbo-rails_app_javascript_turbo_index_js-node_modules_rails_ac-d86665-82aec27bfd70959b3050.js:8773:7788)
	forEach
	q (vendors-node_modules_hotwired_turbo-rails_app_javascript_turbo_index_js-node_modules_rails_ac-d86665-82aec27bfd70959b3050.js:8773:7275)
	O (vendors-node_modules_ahoy_js_dist_ahoy_esm_js-node_modules_animejs_lib_anime_es_js-node_modul-b9f7a7-9fc50a36129a03869516.js:49384:9039)
	map
	message
	```
Not sure how you would coerce something into a string if someone does:
`cable_ready.notification(title:"hello", options: :foo)`

Maybe it's better to let it blow up at that point, or to report a specific error that it can't convert it.

Reply:
I went with `safeString`, `safeArray` and `safeObject` because `toString` is already a thing in JS.

With these new methods, we have an opportunity, in @RolandStuder's vision, to potentially do a bit better than just failing with an exception.

Right now, we have
```js
function safeObject (obj) {
  return obj != null ? obj : {}
}
```
Option 1: do nothing, we're done, let people own their mistakes
Option 2: we could test `obj` to verify that it is an `Object`, and if not, we could log an error that tells people they are passing the wrong thing
Option 3: we could test `obj` to verify that it is an `Object`, and if not, we could silently replace the bad value with `{}`
Option 4: we could test `obj` to verify that it is an `Object`, and if not, we could log an error that tells people they are passing the wrong thing AND coerce the value to `{}` so that events fire and etc

What do you both like best? Obviously I'd do the same thing for all three functions.